src/main/scala/viper/gobra/reporting/StatsCollector.scala

```diff
@@ -0,0 +1,345 @@
+          case GobraMemberInfo(pkgDir, pkg, memberName, args, isTrusted, isAbstract, isImported, false) =>
+            addResult(
+              pkgDir,
+              pkg,
+              memberName,
+              info.pnode.getClass.getSimpleName,
+              args,
+              ViperMemberEntry(e, taskName, time, e.getClass.getSimpleName, success = false, cached = cached, isImported, hasBody = viperMemberHasBody(e)),
+              isTrusted,
+              isAbstract)
+          case _ =>
+        }
+      case _ =>
+    }
+    // Pass message to next reporter
+    reporter.report(msg)
+  }
+
+  def getNumberOfCachedViperMembers: Int = memberMap.values.flatMap(_.viperMembers.values).count(_.cached)
+
+  def getNumberOfCacheableViperMembers: Int = memberMap.values.flatMap(_.viperMembers.values).count(member => member.hasBody && member.nodeType == "Method")
+
+  def writeJsonReportToFile(file: File): Unit = {
+    if((file.exists() && file.canWrite) || file.getParentFile.canWrite) {
+      FileUtils.writeStringToFile(file, getJsonReport(false), UTF_8)
+    }
+  }
+
+  def getJsonReport(shorten: Boolean): String = {
+    val json = "[\n" +
+      memberMap.map({ case (key, gobraMember) => s"""  {
+    "id": "$key",
+    "pkgDir": "${gobraMember.pkgDir}",
+    "pkg": "${gobraMember.pkg}",
+    "name": "${gobraMember.pkg}.${gobraMember.memberName}",
+    "args": "${gobraMember.args}",
+    "nodeType": "${gobraMember.nodeType}",
+    "trusted": ${gobraMember.isTrusted},
+    "abstract": ${gobraMember.isAbstract},
+    "viperMembers": [ """ + "\n" +
+          gobraMember.viperMembers.values.map(viperMember => s"""      {
+        "name": "${viperMemberName(viperMember.member)}",
+        "taskName": "${viperMember.taskName}",
+        "nodeType": "${viperMember.nodeType}",
+        "time": ${viperMember.time},
+        "success": ${viperMember.success},
+        "fromImport": ${viperMember.fromImport},
+        "hasBody": ${viperMember.hasBody},
+        "cached": ${viperMember.cached}
+      }""").mkString(", \n") + "\n    ],\n" + s"""    "dependencies": [""" + "\n" +
+        gobraMember.viperMembers.values
+          .flatMap(viperMember => getDependencies(viperMember.member)
+            .flatMap(dep => findGobraMemberByViperMemberName(viperMember.taskName, dep, gobraMember.pkgDir))
+          )
+          // Filter out dependencies on one self, since this information isn't very useful
+          .filter(_ != gobraMember)
+          .map(gobraMemberEntry => "        \"" + gobraMemberKey(gobraMemberEntry.pkgDir, gobraMemberEntry.pkg, gobraMemberEntry.memberName, gobraMemberEntry.args) + "\"")
+          .toSet
+          .mkString(", \n") + "\n    ]\n  }"
+      }).mkString(", \n") + "\n]\n"
+
+    if(shorten) {
+      // Replaces all spaces, except the ones inside of quotes
+      json.replaceAll("\\s+(?=([^\"]*\"[^\"]*\")*[^\"]*$)", "")
+    } else {
+      json
+    }
```

Comment:
when is it desired/useful to have the shortened version instead of the "normal" one?

Reply:
It saves some space, for example, if we compare the output for the `stats_collector` test directory, the normal version uses 17.1KB and the shortened one 12.5KB. This might be relevant if these will be used as artifacts for a GitHub workflow.

Reply:
Nevermind, replacing spaces for large strings this way seems to be actually very slow and the files don't seem to get unreasonably big

Reply:
we can just drop it then